### DISCUSSION TITLE
v2.3.1

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3201] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (Summary)
* **URL automation and VM downloading** (#2670) (thanks @conath) Allows for controlling and downloading VMs with `utm://` URL. Check out #2670 for more information.
* (macOS) Improved cursor/keyboard capturing and handling: permit macOS shortcuts (minimize, full-screen) when not captured, fix various cursor bugs, basic support for VoiceOver and capturing, and more.
* (macOS) Various UI enhancements: export log/args to file (#3056), title bar changes (#3125), exit confirmation works beyond Cmd+Q/W (#3154) (thanks @ktprograms and @conath)

## Changes (v2.3.1)
* Fixed crash when Fn key is pressed (thanks @ktprograms) (#3179)
* Fixed Unicode support in terminal mode (thanks @tie)
* Added additional Windows icons (thanks @conath)
* (macOS) Fixed numpad issues
* (macOS) Correct key mapping for ISO keyboards (#3175)

## Changes (v2.3.0)
* Fixed DNS resolution on IPv6
* Updated zh-Hans localization (thanks @ty-yqs)
* Ensure temporary files are deleted when VM creation is canceled (#3012)
* Support re-generating and customizing MAC address (#2724)
* On PC/Q35 devices, USB mouse and keyboard will be disabled in order to prevent interference with the PS/2 keyboard and mouse that is always added to the machines. Some software such as Windows Installer will not work with the second mice.
* (macOS) Move VM name in main window from title to subtitle, reduces confusion when switching windows (#3099) (thanks @conath)
* (macOS) Show confirmation alert when closing/quitting with the menu or the button (#3103) (thanks @ktprograms)
* (macOS) Error alert (instead of crash) when trying to use unsupported features on < macOS 11.3 (#3118)
* (macOS) Support starting input capture with Ctrl+Opt (#3119)
* (macOS) Reduced UI thread blocking in terminal mode with lots of output spewing (#2404, #2555)
* (macOS) Add alternative hotkey for capture (Cmd+Opt) (#2710)
* (macOS) Unhide cursor when popup alert shows (#3101)

## Known Issues
* Networking can be broken in some instances, we are still investigating the cause. (#3051, #3094)
* Windows 11 installation may randomly freeze and be corrupted (#3130)

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.3.1'>v2.3.1</a>.</em>

**Comment by conath** [comment]
I have tested the following guests on my M1 MacBook Pro (macOS 11.6); system architecture is in braces:

- Mac OS 9 (PowerPC) OK
- DOS (i368 / x86) OK
- Windows 11 for ARM (aarch64 with Hypervisor; "Bridged Network" mode) Install flaky #3194 but likely Windows problem. otherwise OK – runs great, have not tried to get 2D gpu accel working (QXL)
- Ubuntu 20.04 LTS ARM OK: Desktop works, 3D crashes as expected #3053 
- ReactOS (x86_64) OK
- Windows 10 21h1 (x86_64)
